include/eve/detail/function/simd/common/canonical_shuffle_adapter.hpp

```diff
@@ -0,0 +1,218 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+
+template<typename Internal, simd_value T, std::ptrdiff_t G, std::ptrdiff_t... I>
+EVE_FORCEINLINE auto
+canonical_shuffle_adapter_impl_(EVE_SUPPORTS(cpu_), Internal, T, fixed<G>, pattern_t<I...>)
+requires(G > T::size())
+{
+  // NOTE: we catch this specifically because:
+  //  * sfinae failure here is most likely an error that the user will just miss.
+  //  * It is very easy for the user to add this requirement somewhere at the top
+  static_assert(G <= T::size(), "Group sized passed is bigger than a register, very likely a bug");
+}
+
+template<typename Internal, typename T, typename N, std::ptrdiff_t G, std::ptrdiff_t... I>
+EVE_FORCEINLINE auto
+canonical_shuffle_adapter_impl_(EVE_SUPPORTS(cpu_),
+                                Internal                 internal,
+                                logical<eve::wide<T, N>> x,
+                                fixed<G>                 g,
+                                pattern_t<I...>          p)
+requires(abi_t<T, N>::is_wide_logical) && (G <= N::value)
+{
+  auto shuffle_bits = canonical_shuffle_adapter_impl(internal, x.mask(), g, p);
+  if constexpr( std::same_as<decltype(shuffle_bits), no_matching_shuffle_t> )
+  {
+    return no_matching_shuffle;
+  }
+  else { return bit_cast(shuffle_bits, as(x)); }
+}
+
+template<typename Internal, std::ptrdiff_t G, std::ptrdiff_t... I> struct BundleLambda
+{
+  Internal internal;
+
+  auto operator()(auto x) const noexcept
```

Comment:
If this ia traditional Callable, I guess you did it so you can put EVE_FORCEINLINE on it right ?

Reply:
Good catch